api/server/utils/import/importers.js

```diff
@@ -202,7 +202,7 @@ function processConversation(conv, importBatchBuilder, requestUserId) {
     const isCreatedByUser = role === 'user';
     let sender = isCreatedByUser ? 'user' : 'GPT-3.5';
     const model = mapping.message.metadata.model_slug || openAISettings.model.default;
-    if (model === 'gpt-4') {
+    if (model.includes('gpt-4')) {
       sender = 'GPT-4';
```

Comment:
"Technically", it's GPT-4 _and_ GPT-4o :sweat_smile: (ofc I haven't looked up what `model` looks like)

Reply:
this works for when the model is vanilla "gpt-4" as well as "gpt-4o"

Reply:
I see what you mean about differentiating the sender, I'll think about that soon, as that would mean a few other changes app-wide

Reply:
I mean, tbh, I would'be just slapped all `gpt-4-1106-preview` strings into a huge lookup table (and hide the actual model name e.g. under a hover on the username).

At least for the OpenAI models, that go like

* `gpt-4o-2024-05-13`
* `gpt-4-1106-preview`
* `gpt-3.5-turbo`

I'd try e.g.

```js
function extractUsername(modelName) {
    // Check for specific model name patterns and return corresponding usernames
    if (modelName.startsWith('gpt-4o-')) {
        return 'GPT-4o';
    } else if (modelName.startsWith('gpt-4-')) {
        return 'GPT-4';
    } else if (modelName.startsWith('gpt-3.5-turbo')) {
        return 'GPT-3';
    }

    ...
}
```

and/or something "smarter" (which ofc you can support with tests)